flaky TestHTTPSettings

```
=== RUN   TestHTTPSettings/server_settings
    http_datagram_test.go:33: 
        	Error Trace:	/Users/runner/work/quic-go/quic-go/integrationtests/self/http_datagram_test.go:33
        	Error:      	Received unexpected error:
        	            	timeout: no recent network activity
        	Test:       	TestHTTPSettings/server_settings
=== RUN   TestHTTPSettings/client_settings
--- FAIL: TestHTTPSettings (5.00s)
    --- FAIL: TestHTTPSettings/server_settings (5.00s)
    --- PASS: TestHTTPSettings/client_settings (0.00s)

--- a/integrationtests/self/http_datagram_test.go
+++ b/integrationtests/self/http_datagram_test.go
@@ -6,6 +6,7 @@ import (
 	"encoding/binary"
 	"fmt"
 	"io"
+	"net"
 	"net/http"
 	"net/url"
 	"testing"
@@ -24,9 +25,10 @@ func TestHTTPSettings(t *testing.T) {
 	t.Run("server settings", func(t *testing.T) {
 		tlsConf := tlsClientConfigWithoutServerName.Clone()
 		tlsConf.NextProtos = []string{http3.NextProtoH3}
-		conn, err := quic.DialAddr(
+		conn, err := quic.Dial(
 			context.Background(),
-			fmt.Sprintf("localhost:%d", port),
+			newUDPConnLocalhost(t),
+			&net.UDPAddr{IP: net.IPv4(127, 0, 0, 1), Port: port},
 			tlsConf,
 			getQuicConfig(nil),
 		)
@@ -63,6 +65,7 @@ func TestHTTPSettings(t *testing.T) {
 			EnableDatagrams:    true,
 			AdditionalSettings: map[uint64]uint64{1337: 42},
 		}
+		addDialCallback(t, tr)
 		defer tr.Close()
 		req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("https://localhost:%d/settings", port), nil)
 		require.NoError(t, err)
@@ -99,7 +102,15 @@ func dialAndOpenHTTPDatagramStream(t *testing.T, addr string) *http3.RequestStre
 	tlsConf.NextProtos = []string{http3.NextProtoH3}
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 	defer cancel()
-	conn, err := quic.DialAddr(ctx, u.Host, tlsConf, getQuicConfig(&quic.Config{EnableDatagrams: true}))
+	serverAddr, err := net.ResolveUDPAddr("udp4", u.Host)
+	require.NoError(t, err)
+	conn, err := quic.Dial(
+		ctx,
+		newUDPConnLocalhost(t),
+		serverAddr,
+		tlsConf,
+		getQuicConfig(&quic.Config{EnableDatagrams: true}),
+	)
 	require.NoError(t, err)
 	t.Cleanup(func() { conn.CloseWithError(0, "") })
 

